src/pages/ListSwaps/SwapRow/SwapRow.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Button,
+  CircularProgress,
+  TableCell,
+  TableRow
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
+import React, { useEffect, useReducer } from "react";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import { EmbeddedRepresentationSubEntity } from "../../../../gen/siren";
+import executeAction from "../../../api/executeAction";
+import { Asset, Properties, toMainUnit } from "../../../api/swapTypes";
+import Dialog from "../../../components/Dialog";
+import ExternalLink from "../../../components/ExternalLink";
+import LedgerActionDialogBody from "../LedgerActionDialogBody";
+import SirenActionParametersDialogBody from "../SirenActionParametersDialogBody";
+import SwapStatusIcon from "../SwapStatusIcon";
+import {
+  actionButtonClicked,
+  actionFailed,
+  actionSuccessful,
+  closeLedgerActionDialog,
+  closeSirenParametersDialog,
+  sirenParameterDialogSubmitted
+} from "./events";
+import { ActionExecutionStatus, initialState, reducer } from "./reducer";
+
+interface AssetCellProps {
+  asset: Asset;
+}
+
+function AssetCell({ asset }: AssetCellProps) {
+  return <span>{toMainUnit(asset)}</span>;
+}
+
+const useStyles = makeStyles(() => ({
+  tableRow: {
+    cursor: "pointer"
+  }
+}));
+
+interface SwapRowProps extends RouteComponentProps {
+  swap: EmbeddedRepresentationSubEntity;
+  reload: () => void;
+}
+
+function SwapRow({ swap, history, reload }: SwapRowProps) {
+  const [
+    {
+      state: {
+        actionExecutionStatus,
+        activeLedgerActionDialog,
+        activeSirenParameterDialog
+      },
+      sideEffect
+    },
+    dispatch
+  ] = useReducer(reducer, initialState);
+
+  useEffect(() => {
```

Comment:
:+1: Nice use of `useEffect`!

Reply:
Bluntly stolen from Twitter. Will have to search my history to find it :D

Reply:
https://twitter.com/DavidKPiano/status/1121700680291553281